task_manager/task_manager.cc

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+#include "task_manager.hh"
+#include "seastar/coroutine/parallel_for_each.hh"
+
+logging::logger tmlogger("task manager");
+
+future<> task_manager::invoke_on_task(task_id id, std::function<future<> (task_manager::foreign_task_ptr)> func) {
+    auto& tms = container();
+    task_manager::foreign_task_ptr task = co_await task_manager::lookup_task_on_all_shards(tms, id);
+    co_await tms.invoke_on(task.get_owner_shard(), [func = std::move(func), t = std::move(task)] (auto& tm) mutable {
+        return func(std::move(t));
+    });
+}
```

Comment:
What I had in mind is to follow `lookup_task_on_all_shards` algorithm and call `func` on the right shard while looking for `task_id`.  Also, we can define is as a template so the function can return stuff, like the task sequence_number.

E.g.
```c++
template <typename T>
future<T> task_manager::module::invoke_on_task(task_id id, std::function<future<T> (task_manager::task_ptr)> func) const;
```

It can use the module's `_tm.container()` to get to all shards to lookup `task_id` and if found, invoke `func` on it, returning whatever `func` returns. Otherwise, return exceptional future hold a "task not found" exception.

Reply:
If the module is not available, it could be a static `task_manager` method, like `lookup_task_on_all_shards` that can get the `sharded<task_manager>` as parameter.